### DISCUSSION TITLE
Navigate 进行关联查询时生成的sql丢失on后面的条件，如下图

### DISCUSSION THREAD

**Discussion by landonzeng** [dotnetcore/FreeSql#911] [Q&A]
Navigate 进行关联查询时生成的sql丢失on后面的条件，如下图
![image](https://user-images.githubusercontent.com/7336781/138267226-bb62e504-d84a-49a7-bb5e-3cf4c6729549.png)
[t_device.docx](https://github.com/dotnetcore/FreeSql/files/7388565/t_device.docx)
[t_device_type.docx](https://github.com/dotnetcore/FreeSql/files/7388566/t_device_type.docx)
[t_gw_gateway.docx](https://github.com/dotnetcore/FreeSql/files/7388567/t_gw_gateway.docx)
[t_point.docx](https://github.com/dotnetcore/FreeSql/files/7388568/t_point.docx)
[t_space.docx](https://github.com/dotnetcore/FreeSql/files/7388569/t_space.docx)
[t_sys_type.docx](https://github.com/dotnetcore/FreeSql/files/7388570/t_sys_type.docx)
[DeviceService.docx](https://github.com/dotnetcore/FreeSql/files/7388571/DeviceService.docx)

**Comment by 2881099** [comment]
实体间关联的属性类型不同，所以ON是空的。你先这样查询吧

```c#
    .Include(it => it.Obj_t_sys_type)
    .Include(it => it.Obj_t_space)
    .LeftJoin(it => it.Obj_t_device_type.dt_identifier == it.dt_identifier)
    .Include(it => it.Obj_t_gw_gateway)
```

![JD I)@LV7$D45WIT$6F~EG4](https://user-images.githubusercontent.com/16286519/138270946-6604fca3-66c2-40ea-beba-ee20d2dd4118.png)

![6LQ1AN45}2V5N`YUVINF@_8](https://user-images.githubusercontent.com/16286519/138270951-e47d4e97-d103-4701-b083-a2b744c4b064.png)
